### PR TITLE
TMA-732: all sync_multiple actions fail when filter set is empty

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 1]
```diff
@@ -154,6 +154,8 @@ def call(params)
                 filters << row.to_hash
               end
             end
+            fail 'The filter set can not be empty when using sync_multiple_projects_* mode as the filters contain the project ids \
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'fail 'the filter set can not be empty when using sync_multiple_projects_* mode as the filters contain the project ids \; in which the permissions should be changed'' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Style/MultilineIfModifier: Favor a normal if-statement over a modifier clause in a multiline statement.

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 2]
```diff
@@ -154,6 +154,8 @@ def call(params)
...
+                  in which the permissions should be changed' if filters.empty?
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'filters.empty?' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Tests 'filters.empty?' at least 3 times (RepeatedConditional)

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 3]
```diff
@@ -186,7 +188,8 @@ def call(params)
                 filters << row.to_hash
               end
             end
-            fail 'The filter set can not be empty when using sync_multiple_projects_based_on_custom_id mode' if filters.empty?
+            fail 'The filter set can not be empty when using sync_multiple_projects_* mode as the filters contain the project ids \
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Style/MultilineIfModifier: Favor a normal if-statement over a modifier clause in a multiline statement.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/815/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/682/ : FAILURE in 2m 12s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/943/ : SUCCESS in 2m 44s

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/816/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/683/ : FAILURE in 1m 23s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/944/ : FAILURE in 56s

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 4]
```diff
@@ -154,6 +154,10 @@ def call(params)
                 filters << row.to_hash
               end
             end
+            if filters.empty?
+              fail 'The filter set can not be empty when using sync_multiple_projects_* mode as the filters contain \
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'fail 'the filter set can not be empty when using sync_multiple_projects_* mode as the filters contain \; the project ids in which the permissions should be changed'' 2 times (DuplicateMethodCall)

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 5]
```diff
@@ -154,6 +154,10 @@ def call(params)
                 filters << row.to_hash
               end
             end
+            if filters.empty?
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'filters.empty?' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Tests 'filters.empty?' at least 3 times (RepeatedConditional)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/818/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/685/ : SUCCESS in 37s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/946/ : SUCCESS in 49s

**Pull request merged**